src/Razor/src/Microsoft.VisualStudio.Editor.Razor/Snippets/SnippetInfo.cs

```diff
@@ -39,13 +41,16 @@ internal record SnippetInfo
         => _parsedXmlSnippet.Value;
 }
 
-internal record SnippetCompletionData(string Path)
+internal record SnippetCompletionData([property: JsonProperty(SnippetCompletionData.PropertyName)] string Path)
 {
+    private const string PropertyName = "__razor_snippet_path";
+
     internal static bool TryParse(object? data, [NotNullWhen(true)] out SnippetCompletionData? snippetCompletionData)
     {
         snippetCompletionData = data as SnippetCompletionData;
 
-        if (data is JObject jObject)
+        if (data is JObject jObject &&
+            jObject.Property(PropertyName) is not null)
```

Comment:
Just calling `ToObject` like below will happily return a non-null instance, but with all properties being null if they don't match the Json object. Essentially this means this method never returned null.

I make the property name unique just in case, and I could have made it required and let the try..catch deal with the deserialization failure, but checking the property will be much quicker.

Reply:
😡